cpp/src/arrow/ipc/reader.cc

```diff
@@ -636,8 +637,12 @@ Result<std::shared_ptr<RecordBatch>> LoadRecordBatchSubset(
                             arrow::internal::SwapEndianArrayData(filtered_column));
     }
   }
-  return RecordBatch::Make(std::move(filtered_schema), metadata->length(),
-                           std::move(filtered_columns));
+  auto batch = RecordBatch::Make(std::move(filtered_schema), metadata->length(),
+    std::move(filtered_columns));
+  if (context.options.ensure_memory_alignment) {
+    return util::EnsureAlignment(batch, arrow::util::kValueAlignment, default_memory_pool());
```

Comment:
Can we use the memory pool in context.options.memory_pool?

Reply:
Sure! Ideally we should use the buffer's memory manager rather than the default CPU manager:
https://github.com/apache/arrow/blob/5ad0b3e36f9302c4cf8dd5ab997f30bfab95e2d4/cpp/src/arrow/memory_pool.cc#L907-L916